gensim/models/fasttext.py

```diff
@@ -31,6 +34,15 @@
     >>> from gensim.models import FastText
     >>>
     >>> model = FastText(common_texts, size=4, window=3, min_count=1, iter=10)
+    >>> sentences = [
+    ...     ['computer', 'artificial', 'intelligence'],
+    ...     ['artificial', 'trees'],
+    ...     ['human', 'intelligence'],
+    ...     ['artificial', 'graph'],
+    ...     ['intelligence'],
+    ...     ['artificial', 'intelligence', 'system']
+    ... ]
+    >>> model.train(sentences, total_examples=len(sentences), epochs=model.epochs)
```

Comment:
Where is this `model.epochs` coming from? The model instantiation above shows no such variable.

Reply:
Yes, the epochs parameter is optional, so not included during instantiation.

Unfortunately, there is also some confusion about its name: the FastText constructor uses iter to specify the number of epochs, whereas the superclass uses the proper name epochs.

Reply:
Hm. If it's optional, let's not use it in `train`. Or if we use it in `train`, let's instantiate it explicitly. This neither-here-nor-there example is confusing ("where is this value come from?").

Regarding `iter` / `epochs` -- can you please rename it to `epochs`, consistently? I remember some discussion around this (cc @menshikh-iv @gojomo ), but can't imagine why we'd want both. At most we could support `iter` for a while as a synonym, but with a clear deprecation warning.

This is a perfect opportunity to clean up some of the API mess, rather keep piling on.

Reply:
I agree regarding the cleanup. My preference would be to leave epochs/iter out of the constructor. The model doesn't need that parameter until training time.

Reply:
Model in Gensim generally allow the `trained_model = Constructor(params_including_training_params)`. So breaking that could be confusing (and a big change to backward incompatibility).

I'm not totally opposed though, especially if we still allow ctr params for a while with "deprecated" warnings. The API needs a clean up, and now is a good time.

Reply:
It's not just training parameters you need to include in the constructor. It's also parameters for vocabulary creation. So you're managing at least 3 sets of separate parameters, 2 of which are duplicated by other methods of the class.

Reply:
Yes, we should promote them as separate steps. Question is, do we deprecate (certainly not *remove*) them from ctr?

Reply:
I think the first step should be to deprecate them. After a while, we can remove them, perhaps in time for a major release.

If we want a one-liner way to instantiate and train, we can always write a pure function and promote that.